src/app/dataset-view/additional-components/dataset-settings-component/tabs/access/dataset-settings-access-tab/add-people-modal/add-people-modal.component.ts

```diff
@@ -0,0 +1,123 @@
+import { MaybeNull } from "src/app/interface/app.types";
+import { DatasetCollaborationsService } from "../dataset-collaborations.service";
+import { takeUntilDestroyed } from "@angular/core/rxjs-interop";
+import { LoggedUserService } from "src/app/auth/logged-user.service";
+import { ROLE_OPTIONS } from "./add-people-modal.model";
+
+@Component({
+    selector: "app-add-people-modal",
+    templateUrl: "./add-people-modal.component.html",
+    styleUrls: ["./add-people-modal.component.scss"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AddPeopleModalComponent extends BaseComponent implements OnInit {
+    @Input({ required: true }) public datasetBasics: DatasetBasicsFragment;
+    @Input({ required: true }) public collaborator: MaybeNull<AccountWithRole>;
+    public activeCollaboratorsIds: string[] = [];
+    public role: MaybeNull<DatasetAccessRole> = null;
+    public searchPerson: string = "";
+    public selectedCollaborator: MaybeNull<NameLookupResult>;
+    public searching: boolean = false;
+
+    public readonly DatasetAccessRole: typeof DatasetAccessRole = DatasetAccessRole;
+    public readonly SELECT_ROLE_OPTIONS = ROLE_OPTIONS;
+    private readonly DEFAULT_PER_PAGE = 10;
+
+    public activeModal = inject(NgbActiveModal);
+    private cdr = inject(ChangeDetectorRef);
+    private datasetCollaborationsService = inject(DatasetCollaborationsService);
+    private loggedUserService = inject(LoggedUserService);
+
+    public ngOnInit(): void {
+        this.initState();
+    }
+    private initState(): void {
+        if (this.collaborator) {
+            this.role = this.collaborator.role;
+        }
+        this.datasetCollaborationsService
+            .listCollaborators({ datasetId: this.datasetBasics.id })
+            .pipe(takeUntilDestroyed(this.destroyRef))
+            .subscribe((result: AccountWithRoleConnection) => {
+                this.activeCollaboratorsIds = result.nodes.map((node) => node.account.id);
+            });
+    }
+
+    public search: OperatorFunction<string, readonly NameLookupResult[]> = (text$: Observable<string>) =>
+        text$.pipe(
+            debounceTime(AppValues.SHORT_DELAY_MS),
+            distinctUntilChanged(),
+            tap(() => (this.searching = true)),
+            switchMap((term: string) =>
+                this.datasetCollaborationsService.searchCollaborator({
+                    query: term,
+                    filters: {
+                        byAccount: {
+                            excludeAccountsByIds: [
+                                ...this.activeCollaboratorsIds,
+                                this.loggedUserService.currentlyLoggedInUser.id,
+                            ],
+                        },
+                    },
+                    page: 0,
+                    perPage: this.DEFAULT_PER_PAGE,
+                }),
+            ),
+            tap(() => {
+                this.searching = false;
```

Comment:
In case error happens, the reset of `searching` flag should rather be in `finalize` block

Reply:
Fixed